classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,408 @@
+                    ],
+                    [
+                        'orderBy' => 'i.number',
+                        'direction' => self::ORDER_DIR_ASC
+                    ]
+                ];
+                break;
+            default:
+                $this->resultOrderings = [
+                    [
+                        'orderBy' => 'i.date_published',
+                        'direction' => self::ORDER_DIR_DESC
+                    ]
+                ];
+                break;
+        }
+        return $this;
+    }
+
+    /**
+     * Set published filter
+     *
+     * @return $this
+     */
+    public function filterByPublished(bool $isPublished): self
+    {
+        $this->isPublished = $isPublished;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $volumes
+     *
+     * @return $this
+     */
+    public function filterByVolumes(array $volumes): self
+    {
+        $this->volumes = $volumes;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $numbers
+     *
+     * @return $this
+     */
+    public function filterByNumbers(array $numbers): self
+    {
+        $this->numbers = $numbers;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $years
+     *
+     * @return $this
+     */
+    public function filterByYears(array $years): self
+    {
+        $this->years = $years;
+        return $this;
+    }
+
+    /**
+     * set urlPath filter
+     *
+     * @return $this
+     */
+    public function filterByUrlPath(string $urlPath): self
+    {
+        $this->urlPath = $urlPath;
+        return $this;
+    }
+
+    /**
+     * Set titles filter
+     *
+     * @return $this
+     */
+    public function filterByTitles(array $titles): self
+    {
+        $this->titles = $titles;
+        return $this;
+    }
+
+    /**
+     * Set query search phrase
+     *
+     * @return $this
+     */
+    public function searchPhrase(string $phrase): self
+    {
+        $this->searchPhrase = $phrase;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     *
+     * @return $this
+     */
+    public function limit(?int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     *
+     * @return $this
+     */
+    public function offset(?int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $q = DB::table($this->dao->table, 'i')
+            ->select('i.*')
+            ->leftJoin('custom_issue_orders as o', 'o.issue_id', '=', 'i.issue_id');
+
+        // Issue titles (exact matches)
+        foreach ($this->titles as $title) {
```

Comment:
I think this could be done with a single join, rather than one join per `title`:
```SQL
WHERE i.issue_id IN (SELECT issue_id FROM issue_settings WHERE setting_name='title' AND setting_value IN (?, ?, ?)
```